airflow/jobs.py

```diff
@@ -96,25 +96,54 @@ class BaseJob(Base, LoggingMixin):
         Index('idx_job_state_heartbeat', state, latest_heartbeat),
     )
 
+    heartrate = conf.getfloat('scheduler', 'JOB_HEARTBEAT_SEC'),
+
     def __init__(
             self,
             executor=executors.get_default_executor(),
-            heartrate=conf.getfloat('scheduler', 'JOB_HEARTBEAT_SEC'),
+            heartrate=None,
             *args, **kwargs):
         self.hostname = get_hostname()
         self.executor = executor
         self.executor_class = executor.__class__.__name__
         self.start_date = timezone.utcnow()
         self.latest_heartbeat = timezone.utcnow()
-        self.heartrate = heartrate
+        if heartrate is not None:
+            self.heartrate = heartrate
```

Comment:
@ashb , the condition here should be `heartrate is None` rather than `heartrate is not None`, right?

Reply:
@XD-DENG No I don't think so. If the `hearteate` parameter is passed in (i.e. `not None`) then we want to use that provided value, otherwise we leave the existing class-level default (set on L99):


```python-console
>>> from airflow.jobs import BaseJob
>>> j1 = BaseJob()
>>> j1.heartrate
5.0
>>> j2 = BaseJob(heartrate=1.0)
>>> j2.heartrate
1.0
```

Reply:
Yes, you're right. my bad